dali/ft/filecopy.cpp

```diff
@@ -3279,6 +3276,31 @@ bool FileSprayer::disallowImplicitReplicate()
 
 }
 
+void FileSprayer::checkPushSupported()
+{
+    if (targetSupportsConcurrentWrite)
+        return; // push ok
+
+    bool multilpeSourcesPerTarget = false;
+    std::vector<bool> targetUsed;
+    targetUsed.resize(targets.ordinality());
+    ForEachItemIn(idx, partition)
+    {
+        PartitionPoint & cur = partition.item(idx);
+        if (targetUsed[cur.whichOutput])
+        {
+            multilpeSourcesPerTarget = true;
+            break;
+        }
+        targetUsed[cur.whichOutput] = true;
+    }
+    if (!multilpeSourcesPerTarget)
+        return; // push ok
+
+    IWARNLOG("Forcing pull. Multiple source partitions write to same target, and target does not support concurrent write");
+    cachedUsePull = true;
```

Comment:
cleaner to return a boolean and only update cachedUsePull inside usePullOperation()